### PR TITLE
Items can now be added to cart

### PR DESCRIPTION
**Pull request by ydaniju**
Why?
Users should be able to add products to cart

How?
- Get product attributes from form on clicking 'add to cart'
- Save attributes into local storage
- Update DOM elements with information in local storage

https://www.pivotaltracker.com/story/show/110060166
[Finishes #110060166]

### CODE REVIEW

**Comment by makinwab** [comment]
1. There should be a default / blank state for the cart dropdown. If the cart is empty, there should be a label like **Your cart is empty** and the **proceed to checkout** button should be removed.
2. The number of items beside the **cart** link can be positioned and designed to look better. Also, it should not be there if there are no items in the cart.
3. The number beside the **cart** link at the moment depicts the total _quantity_ of each item in the cart. This should not be so. It should be the total number of items added to cart.

**Comment by ydaniju** [comment]
Very well @andela-bmakinwa. Corrections are on the way.

On Thu, Jan 28, 2016 at 6:55 PM, Makinwa Olubukola <notifications@github.com

> wrote:
> 
>    1.
> 
>    There should be a default / blank state for the cart dropdown. If the
>    cart is empty, there should be a label like _Your cart is empty_ and
>    the _proceed to checkout_ button should be removed.
>    2.
> 
>    The number of items beside the _cart_ link can be positioned and
>    designed to look better. Also, it should not be there if there are no items
>    in the cart.
>    3.
> 
>    The number beside the _cart_ link at the moment depicts the total
>    _quantity_ of each item in the cart. This should not be so. It should
>    be the total number of items added to cart.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/andela/getmyshop/pull/40#issuecomment-176304853.

## 

_Best Regards,_

_Yusuf Daniju /_\* Andela.com
http://andela.com/?utm_source=andela&utm_medium=email&utm_campaign=signature *
_Software Developer / *_+234.807.266.7248 <+234.807.266.7248>*\* / @Andela
http://bit.ly/andelatwitter*

*"The startup that's harder to get into than Harvard" - *CNN
http://snip.ly/HfId
See what The Wall Street Journal http://snip.ly/u9AI has to say.

**Comment by oluosiname** [comment]
Nice work @andela-ydaniju. It would be good to have the unit of the total price indicated in the cart drop down.

**Comment by ydaniju** [comment]
Thanks @andela-oosiname. Your feedback would be incorporated now. More?

On Mon, Feb 1, 2016 at 10:04 AM, Olumuyiwa Osiname <notifications@github.com

> wrote:
> 
> Nice work @andela-ydaniju https://github.com/andela-ydaniju. It would
> be good to have the price unit in the cart drop down.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/andela/getmyshop/pull/40#issuecomment-177863952.

## 

_Best Regards,_

_Yusuf Daniju /_\* Andela.com
http://andela.com/?utm_source=andela&utm_medium=email&utm_campaign=signature *
_Software Developer / *_+234.807.266.7248 <+234.807.266.7248>*\* / @Andela
http://bit.ly/andelatwitter*

*"The startup that's harder to get into than Harvard" - *CNN
http://snip.ly/HfId
See what The Wall Street Journal http://snip.ly/u9AI has to say.

**Comment by ChigboIO** [comment]
Great work @andela-ydaniju, I should come for some tutorial on JavaScript.

Since we are limiting the user to the available number of a given product for the quantity, I think we should also keep track of that while adding product to cart.
We can keep a front end variable off the available quantity to check it with.

**Comment by 0duaht** [comment]
Nice work, @andela-ydaniju. This is top-notch.

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -2,28 +2,7 @@
...
-$button-color: #f3f3f3;
-$section-font: 'ProximaNova', sans-serif, Arial;
-$shadow-color: rgba(0, 0, 0, 0.75);
-
-$facebook-blue: #3b5998;
-$twitter-blue: #4099ff;
-$instagram-red: #3f729b;
-$pinterest-red: #c92228;
-
+@import "variables";
```
Comment (ChigboIO) [inline app/assets/stylesheets/application.scss:5]: Good job moving these variables to another file :+1:

File: app/views/products/show.html.erb [hunk 2]
```diff
@@ -59,22 +59,31 @@
 
         </div>
       </div>
-      <div class="buy-now">
-        <div class="row">
-          <form method="get">
+      <input  type=hidden>
```
Comment (ChigboIO) [inline app/views/products/show.html.erb:62]: Is this hidden field used for anything?

File: app/views/products/show.html.erb [hunk 3]
```diff
@@ -59,22 +59,31 @@
...
             <div class="col s1 bold">Qty: </div>
-            <div class="col s2"><input type="number" value="1" min = 1 max=<%= @product.quantity %> class="products-input"></div>
-            <div class="col s4"><input type="submit" class="waves-effect btn products-input" value="BUY NOW" /></div>
+            <div class="col s2"><%= number_field_tag :quantity, 1, in: 1..@product.quantity, step: 1 %>
+            </div>
+            <div class="col s4"><%= submit_tag "ADD TO CART", class: "waves-effect btn products-input", id: "submit"%></div>
             <div class="col s5">
               <a href="#" title="" class="wish-list-link">
-                <i class="fa fa-heart-o wishlist-icon"></i> Add to Wish List
+                <i class="fa fa-heart-o wishlist-icon"s></i> Add to Wish List
```
Comment (ChigboIO) [inline app/views/products/show.html.erb]: I found a stray `s` in the `i` tag. Wondering if it is doing anything in the ministry. :smile:

**Comment by ydaniju** [comment]
Thank @andela-toduah and @andela-echigbo. As per the front-end validation, I think that is important at the point of checkout. Adding to cart may be done by 50 people at the same time. The first to reach our `database` gets the pie.

File: app/views/products/show.html.erb (hunk 2 above)
Comment (ydaniju) [inline app/views/products/show.html.erb:62]: Yes @EmmanuelScience, it is in use.

File: app/assets/javascripts/cart.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    var items = localStorage !== null ? localStorage[this.cartName + "_items"] : null;
+    if (items !== null && JSON !== null) {
+        try {
+            items = JSON.parse(items);
+            for (var i = 0; i < items.length; i++) {
+                var item = items[i];
+                if (item.code !== null && item.name !== null && item.price !==
+                    null && item.quantity !== null &&
+                        item.image !== null) {
+                    item = new CartItem(
```
Comment (ChigboIO) [inline app/assets/javascripts/cart.js]: I think we need to have the `id` of the item, some we can easily recognize the item at the backend and get more information about the item

Comment (ydaniju) [inline app/assets/javascripts/cart.js]: Yeah. It just wasn't important for this feature. I guess other features
depend on it.

On Mon, Feb 1, 2016 at 3:54 PM, Emmanuel Chigbo notifications@github.com
wrote:

> In app/assets/javascripts/cart.js
> https://github.com/andela/getmyshop/pull/40#discussion_r51426779:
> 
> > -        self.saveItems();
> > -        self.clearCart = false;
> > -    });
> >   +}
> >   +
> >   +shoppingCart.prototype.loadItems = function () {
> > -    var items = localStorage !== null ? localStorage[this.cartName + "_items"] : null;
> > -    if (items !== null && JSON !== null) {
> > -        try {
> > -            items = JSON.parse(items);
> > -            for (var i = 0; i < items.length; i++) {
> > -                var item = items[i];
> > -                if (item.code !== null && item.name !== null && item.price !==
> > -                    null && item.quantity !== null &&
> > -                        item.image !== null) {
> > -                    item = new CartItem(
> 
> I think we need to have the id of the item, some we can easily recognize
> the item at the backend and get more information about the item
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/andela/getmyshop/pull/40/files#r51426779.

## 

_Best Regards,_

_Yusuf Daniju /_\* Andela.com
http://andela.com/?utm_source=andela&utm_medium=email&utm_campaign=signature *
_Software Developer / *_+234.807.266.7248 <+234.807.266.7248>*\* / @Andela
http://bit.ly/andelatwitter*

*"The startup that's harder to get into than Harvard" - *CNN
http://snip.ly/HfId
See what The Wall Street Journal http://snip.ly/u9AI has to say.

**Pull request merged**